samples/cpp/benchmark_app/inputs_filling.cpp

```diff
@@ -117,6 +117,20 @@ void fillBlobBinary(Blob::Ptr& inputBlob,
                     const size_t& inputId,
                     const size_t& inputSize) {
     MemoryBlob::Ptr minput = as<MemoryBlob>(inputBlob);
+    auto adjBatchSize = batchSize;
+
+    // Check layout
+    std::stringstream ss;
+    auto tensorDesc = inputBlob->getTensorDesc();
+    ss << tensorDesc.getLayout();
+    auto layout = ss.str();
+    std::size_t batchIndex = layout.find("N");
+    if (batchIndex == std::string::npos) {
+        adjBatchSize = 1;
+    } else if (tensorDesc.getDims().at(batchIndex) != batchSize) {
+        adjBatchSize = tensorDesc.getDims().at(batchIndex);
+    }
+
```

Comment:
Shouldn't we reshape network using new provided batch size instead ?

Reply:
It is not reshaped due to the following code.
samples/cpp/benchmark_app/utils.hpp:111 ~ 118
        // Update shape with batch if needed
        if (batch_size != 0) {
            std::size_t batch_index = info.layout.find("N");
            if ((batch_index != std::string::npos) && (info.shape.at(batch_index) != batch_size)) {
                info.shape[batch_index] = batch_size;
                reshape_required = true;
            }
        }
Its layout is only 'C'. Do you think we should modify this code to reshape 'C' layout tensor to 'NC'?